src/zone/timer.cpp

```diff
@@ -74,30 +74,30 @@ bool TimersScheduler::StartMainLoop() {
                 return;
             }
 
-            const auto& topTimer = activeTimers.top();
-            if (topTimer.expirationTime <= std::chrono::high_resolution_clock::now()) {
-                if (timers[topTimer.index].active) {
+            auto nextExpirationTime = activeTimers.top().expirationTime;
+            if (nextExpirationTime <= std::chrono::high_resolution_clock::now()) {
+                // Pop before callback）to re-arm the timer.
+                auto expiredTimer = activeTimers.top();
+                activeTimers.pop();
+
+                if (timers[expiredTimer.index].active) {
+                    timers[expiredTimer.index].active = false;
+
                     try {
-                        // Fire the callback.
-                        // The callback is assumed to be very fast as it is meant to dispatch to appropriate
-                        // callback queues.
-                        timers[topTimer.index].callback();
+                        // Fire the callback. The callback is assumed to be 
```

Comment:
Revise the comments here.

The comments are mainly about **why the change should be done in this commit**, but not explain the logic and behavior.

Also, this will not be fast since the new code tries to create a task object in `TimerWrap::SetTimersCallback`.

Reply:
revert back comments. Also create task is still fast enough. it just create several object.